tests/Stripe/BaseStripeClientTest.php

```diff
@@ -220,4 +220,74 @@ public function testRequestWithNoVersionDefaultsToPinnedVersion()
             []
         );
     }
+
+    public function testSetClientAppInfo()
+    {
+        $appInfo = [
+            'name' => 'MyTestApp',
+            'version' => '1.2.34',
+            'url' => 'https://mytestapp.example',
+            'appPartnerId' => 'partner_1234',
+        ];
+
+        $client = new BaseStripeClient([
+            'api_key' => 'sk_test_appinfo',
+            'api_base' => MOCK_URL,
+            'app_info' => $appInfo,
+        ]);
+
+        $this->expectsRequest('get', '/v1/charges/ch_123', null, [
+            'User-Agent: ' . 'Stripe/v1 PhpBindings/' . Stripe::VERSION . ' MyTestApp/1.2.34 (https://mytestapp.example)',
+        ]);
+        $charge = $client->request('get', '/v1/charges/ch_123', [], []);
+    }
+
+    public function testSetClientAppInfoOnlyName()
+    {
+        $client = new BaseStripeClient([
+            'api_key' => 'sk_test_appinfo',
+            'api_base' => MOCK_URL,
+            'app_info' => [
+                'name' => 'MyTestApp',
+            ],
+        ]);
+
+        $this->expectsRequest('get', '/v1/charges/ch_123', null, [
+            'User-Agent: ' . 'Stripe/v1 PhpBindings/' . Stripe::VERSION . ' MyTestApp',
+        ]);
+        $charge = $client->request('get', '/v1/charges/ch_123', [], []);
+    }
+
+    public function testClientAppInfoFallsBackToGlobal()
+    {
+        Stripe::setAppInfo('MyTestApp', '1.2.34', 'https://mytestapp.example');
+        $client = new BaseStripeClient([
+            'api_key' => 'sk_test_appinfo',
+            'api_base' => MOCK_URL,
+        ]);
+
+        $this->expectsRequest('get', '/v1/charges/ch_123', null, [
```

Comment:
Instead of `expectsRequest`, can we stub the actual HTTP request by mocking the HTTP client like in https://github.com/stripe/stripe-php/blob/beta/tests/Stripe/BaseStripeClientTest.php#L415? I think this will also give us more control when asserting the headers so we can test `X-Stripe-Client-User-Agent`.